### ISSUE TITLE
Routing Rules do not allow dynamic ports

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#321]
We're using mountebank to mock APIs. This leads to quite some routing rules, because each imposter is hosted on its own port.

So when having two imposters, the routing rules look like this:
```json
{
  "/mountebank-internal/imposters/4553/(.*)": {
    "description": "Mountebank imposter for API X",
    "url": "http://localhost:4553/$1"
  },
  "/mountebank-internal/imposters/4999/(.*)": {
    "description": "Mountebank imposter for API Y",
    "url": "http://localhost:4999/$1"
  }
}
```
These two (and lots more) rules could be easily merged into a single routing rule:
```json
{
  "/mountebank-internal/imposters/(4553|4999)/(.*)": {
    "description": "Mountebank imposter for API X and API Y",
    "url": "http://localhost:$1/$2"
  }
}
```
Currently, this merged routing rule is not possible because the regex pattern in the [RuleFactory](https://github.com/swisspush/gateleen/blob/develop/gateleen-routing/src/main/java/org/swisspush/gateleen/routing/RuleFactory.java) class does not allow to define wildcards (like $1) as a port.

### Solution
* Change regex pattern to also allow wildcard values for port configuration 
* Mark corresponding [Rule](https://github.com/swisspush/gateleen/blob/develop/gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Rule.java) to use wildcard values for the port
* In [Forwarder](https://github.com/swisspush/gateleen/blob/develop/gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java) class extract dynamic port value from current request and use extracted port for the forwarding

**Comment by lbovet** [comment]
Powerful idea. Make sure to review the routes regexps correctly. Someone (if not me) will surely try to netcat on /mountebank-internal/imposters/22/ someday :)

**Comment by mcweba** [comment]
Released in [v1.1.55](https://github.com/swisspush/gateleen/releases/tag/v1.1.55)
